osb.go

```diff
@@ -0,0 +1,188 @@
+}
+
+func (b broker) Services(ctx context.Context) ([]brokerapi.Service, error) {
+	var services []brokerapi.Service
+	for _, s := range b.c.Config.Catalog.Services {
+		var servicePlans []brokerapi.ServicePlan
+		for _, p := range s.Plans {
+			sp := brokerapi.ServicePlan{
+				ID:          p.ID,
+				Name:        p.Name,
+				Description: p.Description,
+				Free:        &p.Free,
+				Bindable:    &p.Bindable,
+			}
+			servicePlans = append(servicePlans, sp)
+		}
+		dc := brokerapi.Service{
+			Name:                 s.Name,
+			ID:                   s.ID,
+			Description:          s.Description,
+			Tags:                 s.Tags,
+			Bindable:             s.Bindable,
+			InstancesRetrievable: s.InstancesRetrievable,
+			BindingsRetrievable:  s.BindingsRetrievable,
+			PlanUpdatable:        s.PlanUpdateable,
+			Plans:                servicePlans,
+		}
+		services = append(services, dc)
+	}
+	return services, nil
+}
+
+func (b broker) Provision(ctx context.Context, instanceID string, details brokerapi.ProvisionDetails, asyncAllowed bool) (brokerapi.ProvisionedServiceSpec, error) {
+	// FIXME: Uncomment after prefixing is done
+	// if err := ValidInstanceID(instanceID); err != nil {
+	// 	return brokerapi.ProvisionedServiceSpec{}, err
+	// }
+	plan, err := b.c.Config.Catalog.FindPlan(details.ServiceID, details.PlanID)
+	if err != nil {
+		return brokerapi.ProvisionedServiceSpec{}, err
+	}
+
+	var params map[string]interface{}
+	err = json.Unmarshal(details.RawParameters, &params)
+	if err != nil {
+		return brokerapi.ProvisionedServiceSpec{}, err
+	}
+
+	_, err = b.c.Provision(&Instance{
+		ID:             instanceID,
+		Plan:           plan,
+		Root:           b.c.Root,
+		Prefix:         b.c.Config.Prefix,
+		VaultPrefix:    b.c.Config.Vault.Prefix,
+		UserParameters: params,
+	})
+
+	return brokerapi.ProvisionedServiceSpec{
+		IsAsync:       true,
+		AlreadyExists: false,
+	}, err
+}
+
+func (b broker) Deprovision(ctx context.Context, instanceID string, details brokerapi.DeprovisionDetails, asyncAllowed bool) (brokerapi.DeprovisionServiceSpec, error) {
+	_, gone, err := b.c.Deprovision(instanceID)
+	if err != nil {
+		return brokerapi.DeprovisionServiceSpec{}, err
+	}
+
+	if gone {
+		return brokerapi.DeprovisionServiceSpec{}, fmt.Errorf("service instance '%s' already deprovisioned", instanceID)
+	}
+
+	return brokerapi.DeprovisionServiceSpec{
+		IsAsync: true,
+	}, nil
+}
+
+func (b broker) GetInstance(ctx context.Context, instanceID string) (brokerapi.GetInstanceDetailsSpec, error) {
+	inst, ok := b.c.instances[instanceID]
+	if !ok {
+		return brokerapi.GetInstanceDetailsSpec{}, fmt.Errorf("service instance '%s' not found", instanceID)
+	}
+
+	return brokerapi.GetInstanceDetailsSpec{
+		PlanID:     inst.Plan.ID,
+		Parameters: inst.UserParameters,
+	}, nil
+}
+
+func (b broker) Update(ctx context.Context, instanceID string, details brokerapi.UpdateDetails, asyncAllowed bool) (brokerapi.UpdateServiceSpec, error) {
+	panic("not implemented")
```

Comment:
This probably shouldn't panic; instead we should return an empty `brokerapi.UpdateServiceSpec` and an error about not implementing / not supporting this endpoint.